frontend/src/components/viewers/VisualizationCreator.tsx

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as React from 'react';
+import BusyButton from '../../atoms/BusyButton';
+import FormControl from '@material-ui/core/FormControl';
+import Input from '../../atoms/Input';
+import InputLabel from '@material-ui/core/InputLabel';
+import MenuItem from '@material-ui/core/MenuItem';
+import Select from '@material-ui/core/Select';
+import Viewer, { ViewerConfig } from './Viewer';
+import { ApiVisualizationType } from '../../apis/visualization';
+
+
+export interface VisualizationCreatorConfig extends ViewerConfig {
+  // Whether there is currently a visualization being generated or not.
+  isBusy?: boolean;
+  // Function to be called when a visualization with the provided arguments,
+  // inputPath, and visualization type needs to be generated.
+  onGenerate?: (visualizationArguments: string, inputPath: string, type: ApiVisualizationType) => void;
+}
+
+interface VisualizationCreatorProps {
+  configs: VisualizationCreatorConfig[];
+  maxDimension?: number;
+}
+
+interface VisualizationCreatorState {
+  arguments: string;
```

Comment:
if these have a particular form (JSON?), it's probably worth mentioning that in a comment, unless you expect it to change